server/route.go

```diff
@@ -1,28 +1,92 @@
 package server
 
 import (
+	"errors"
 	"fmt"
 	"github.com/smancke/guble/protocol"
+	"runtime"
+	"sync"
+	"time"
+)
+
+var (
+	errEmptyQueue = errors.New("Empty queue")
+	errTimeout    = errors.New("Channel sending timeout")
+
+	defaultQueueCap = 50
 )
 
 // NewRoute creates a new route pointer
-func NewRoute(path, applicationID, userID string, channel chan *MessageForRoute) *Route {
-	return &Route{
+func NewRoute(path, applicationID, userID string, c chan *MessageForRoute) *Route {
+	route := &Route{
+		messagesC:     c,
+		queue:         newQueue(),
+		queueSize:     0,
+		timeout:       -1,
+		closeC:        make(chan bool),
 		Path:          protocol.Path(path),
-		messagesC:     channel,
 		UserID:        userID,
 		ApplicationID: applicationID,
 	}
+	return route
 }
 
 // Route represents a topic for subscription that has a channel to receive messages.
 type Route struct {
```

Comment:
I think, all fields here should be documented to get an understanding of the behaviour of the type.